### PR TITLE
Multicore Ingress bug and header_flit

### PR DESCRIPTION
**Pull request by vineetjain07**
This PR contains -
1) Index in $csr_core for NUM_CORES > 1 (without this, Cores>2 were failing)
2) Shift the $header_flit field to {(msb)VC - SRC - DEST(lsb)}
3) Fix for Ingress buffer, by blocking flits for PKTRD(if flit is invalid on its 1st cycle) until it replays with a valid flit.

### CODE REVIEW

File: warp-v.tlv [hunk 1]
```diff
@@ -4456,4 +4469,4 @@ m4+module_def
    m4+warpv_top()
 
 \SV
-   endmodule
+   endmodule
```
Comment (stevehoover) [inline warp-v.tlv:4472]: Probably done by your text editor. You might want to change your editor's setting for this.

File: warp-v.tlv [hunk 2]
```diff
@@ -3742,17 +3742,30 @@ m4+definitions(['
...
+         $negedge_pktrd = ! $do_pktrd &&  >>1$do_pktrd; // last cycle
+
+
       @0
-         // Replay for PKTRD with no data read.
-         $pktrd_blocked = $is_pktrd && ! $trans_valid;
+         // Replay for PKTRD with no data read.(and if 1st cycle of PKTRD flit is invalid)
+
+         $pktrd_blocked = $posedge_pktrd ? ($is_pktrd && ! $trans_valid) : 
+                          $negedge_pktrd ? '0 : $RETAIN;
```
Comment (stevehoover) [inline warp-v.tlv:3755]: $RETAIN makes this state. State should generally have a $reset case. I don't think this needs to be state. I don't actually understand how this expression has any impact. Isn't $posedge_pktrd a pulse? So this expression will pulse only if ! $trans_valid, in which case there's nothing to block?? In any case, the first issue is that a clear comment describing this case is needed.

**Review by stevehoover — CHANGES_REQUESTED** [review]
I don't understand well enough to approve. Needs commenting.

File: warp-v.tlv (hunk 1 above)
Comment (vineetjain07) [reply warp-v.tlv]: done

File: warp-v.tlv (hunk 2 above)
Comment (vineetjain07) [reply warp-v.tlv:3755]: Yes, $posedge_pktrd is a pulse. So, $pktrd_blocked is asserted on $posedge_pktrd only if  ! $trans_valid, and it should remain HIGH until $do_pktrd is low.

**Comment by vineetjain07** [comment]
Fixed - 
1) typo for GoodPathMask in docs.
2) speculative CSR pktrd.
3) VIZ for /bank[*]

**Review by stevehoover — APPROVED** [review]

**Pull request merged**